### DISCUSSION TITLE
URL Shorting with YOURLS not working when using PHP 8 or above on NGINX

### DISCUSSION THREAD

**Discussion by scsa20** [PrivateBin/PrivateBin#1255] [Q&A] [answered]
### Describe the problem/question

When using PHP 8 or above the URL shorting option does not work (even though it still creates a shorten URL), this is the error that I get in the logs

`2024-02-21 01:37:09	Error	70.176.162.161	500	GET /shorten.php?link=https%3A%2F%2Fp.sc20.me%2F%3F67c992ec7ebd613c%23B6ThMAx3yPAygYTQ3RecardZcvw4H68VscEvBWwwJpPC HTTP/2.0		Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36	240	nginx SSL/TLS access
2024-02-21 01:37:09	Error	70.176.162.161	500	GET /shorten.php?link=https%3A%2F%2Fp.sc20.me%2F%3F67c992ec7ebd613c%23B6ThMAx3yPAygYTQ3RecardZcvw4H68VscEvBWwwJpPC HTTP/2.0		Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36	240	nginx SSL/TLS access
2024-02-21 01:37:09	Error	70.176.162.161		2418991#0: *3003 FastCGI sent in stderr: "PHP message: PHP Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, stdClass given in /var/www/vhosts/sc20.me/p.sc20.me/shorten.php:50
Stack trace:
#0 {main}
thrown in /var/www/vhosts/sc20.me/p.sc20.me/shorten.php on line 50" while reading response header from upstream				nginx error
2024-02-21 01:37:09	Error	70.176.162.161		2418991#0: *3003 FastCGI sent in stderr: "PHP message: PHP Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, stdClass given in /var/www/vhosts/sc20.me/p.sc20.me/shorten.php:50
Stack trace:
#0 {main}
thrown in /var/www/vhosts/sc20.me/p.sc20.me/shorten.php on line 50" while reading response header from upstream				nginx error`

It works just fine when the privatebin instance is set to use PHP 7.4.x

### Did you use the FAQ section?

- [X] Yes, I have read [the FAQ](https://github.com/PrivateBin/PrivateBin/wiki/FAQ) and I found no solution/answer there.

### What you did?

1. Set PHP version to 8 (I'm using PHP 8.3.3)
2. Create a paste.
3. Try to create a shorten URL from the screen.

### What happens

The shorten URL does not created but it doesn't get return to the page and is created to a blank page (you must log into YOURLS to grab the shorten link).

### What should happen

Page should refresh with the shorten link like it did when using PHP 7.4

### Additional information

Not sure if this is considered a bug or not but creating a support ticket to see if anyone can help me out with this. I am using Plesk panel as well but made sure all settings are in ordered. It works perfectly fine when the PHP version for the privatebin instance is set to 7.4.33. The YOURLS instance is on 8.3.3 and has no issues.

### Server address

https://p.sc20.me

### Server OS

AlmaLinux 8.9

### Webserver

NGINX

### PrivateBin version

1.7.1

### Browser and version

Brave 1.62.165 Chromium: 121.0.6167.184 (Official Build) (64-bit) 

### Local operating system and version

Windows 11 Version 22H2 (Build 22621.3007)

### Issue reproducibility

No, I cannot reproduce it on https://privatebin.net.

**Comment by elrido** [comment, marked as answer]
The shorten.php is not anything that this project provides, but I found it [mentioned in a discussion of a proposed feature](https://github.com/PrivateBin/PrivateBin/issues/725#issuecomment-744012199), maybe you were using that?

We have since gained functionality for a YOURLS proxy in #997 which became part of [release 1.5](https://github.com/PrivateBin/PrivateBin/blob/master/CHANGELOG.md#15-2022-12-11). You can find more information on how to set it up in the [configuration wiki page](https://github.com/PrivateBin/PrivateBin/wiki/Configuration#yourls) as well as in the comments on the relevant settings in the sample configuration file:

https://github.com/PrivateBin/PrivateBin/blob/03e7f6a9a7a24c5e25caa07d1a0e7366731d8fe3/cfg/conf.sample.php#L250-L260
